### PR TITLE
Include some json schemas within pystac

### LINKED ISSUE #1007: Include some jsonschema documents in the pystac distribution
When doing some testing for #949, I was reminded that _every single_ validate call hits the network. This makes it pretty hard to use PySTAC in a network-denied environment. I don't see any reason why we couldn't bundle the most common schemas in the PySTAC distribution and load them from the local system for validation -- the currently-supported version (v1.0.0 as of today) of the main data structures, plus GeoJSON Feature, seem like obvious candidates. It would additionally prevent issues like #622, where validation broke because the GeoSON Feature schema was a 404.

### PR DESCRIPTION
**Pull request by jsignell**
**Related Issue(s):**

- closes #1007, related to #1162

**Description:**

I pulled a lot of @gadomski's work off the v2 branch, but left the existing validation mechanism in place, just calling out to the new stuff when the schema is local.

**PR Checklist:**

- [x] `pre-commit` hooks pass locally
- [x] Tests pass (run `scripts/test`)
- [x] Documentation has been updated to reflect changes, if applicable
- [x] This PR maintains or improves overall codebase code coverage.
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac/blob/main/CHANGELOG.md). See [the docs](https://pystac.readthedocs.io/en/latest/contributing.html#changelog) for information about adding to the changelog.

### CODE REVIEW

File: pystac/validation/jsonschemas/extensions/eo/v1.1.0/schema.json
Comment (gadomski) [inline pystac/validation/jsonschemas/extensions/eo/v1.1.0/schema.json:1]: I'm less sure that we want to include extension schemas in this library. The extensions release more quickly and with less publicity, and so we get less of a win for more complexity. IMO if folks need performance gains while validating extensions, they should build and hold onto their `JsonSchemaSTACValidator` and its associated cache.

**Review by gadomski — CHANGES_REQUESTED** [review]

File: pystac/validation/local_validator.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        for name in ("basics", "datetime", "instrument", "licensing", "provider"):
+            resolver.store[
+                f"https://schemas.stacspec.org/v{version}/item-spec/json-schema/{name}.json"
+            ] = _read_schema(f"stac-spec/v{version}/{name}.json")
+        return Draft7Validator(schema, resolver=resolver)
+
+
+def _read_schema(file_name: str) -> Dict[str, Any]:
+    with (here / "jsonschemas" / file_name).open("r") as f:
+        return cast(Dict[str, Any], json.load(f))
```
[the remark this answers is not in the document]
Comment (gadomski) [reply pystac/validation/local_validator.py:88]: Yeah, I don't understand all the intricacies, but I _think_ `importlib` tries to reserve the right to lay things out differently, e.g. I think this method support zipped dependencies. I wish I grokked it all better, though 😞.

[the remark this answers is not in the document]
Comment (jsignell) [reply pystac/validation/local_validator.py]: ooh zipping. Yeah that makes sense.

**Review by gadomski — APPROVED** [review]

**Comment by jsignell** [comment]
Oh rats. merge queue is never going to work with that failing schema lookup

**Pull request merged**